### PR TITLE
container: pass StopTimeout to the systemd slice

### PR DESCRIPTION
**Pull request by giuseppe**
#### Does this PR introduce a user-facing change?

```release-note
Now Podman specifies the StopTimeout to systemd, so that it is honored when systemd stops the scope
```

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
@containers/podman-maintainers PTAL

File: libpod/container_internal_common.go [hunk 1]
```diff
@@ -568,6 +568,11 @@ func (c *Container) generateSpec(ctx context.Context) (s *spec.Spec, cleanupFunc
 	g.SetRootPath(c.state.Mountpoint)
 	g.AddAnnotation("org.opencontainers.image.stopSignal", strconv.FormatUint(uint64(c.config.StopSignal), 10))
 
+	if c.config.StopTimeout != 0 {
+		annotation := fmt.Sprintf("uint64 %d", c.config.StopTimeout*1000000) // sec to usec
+		g.AddAnnotation("org.systemd.property.TimeoutStopUSec", annotation)
```
Comment (Luap99) [inline libpod/container_internal_common.go:573]: So I assume the oci runtime sets this on the slice then? Does work with both crun and runc?
Also in this case do we also have to set the stop signal we use or is this already done?

**Review by Luap99 — COMMENTED** [review]
You commit message says `systemd slice` in the title but is sets it on the scope.
Is that correct or should it say scope?

File: libpod/container_internal_common.go (hunk 1 above)
Comment (giuseppe) [reply libpod/container_internal_common.go:573]: it is on the scope.  I'll fix the git commit message.  It works both with crun and runc, and if an OCI runtime doesn't support it then it is just ignored.

I don't see a way to override the stop signal using the same mechanism

Comment (Luap99) [reply libpod/container_internal_common.go:573]: Ok I took a quick look and testing it we do not pass down the proper stop signal to the scope which means if the container doesn't react on SIGTERM (and uses a different signal) the fix will not help.

This should do it, mind adding it here as well and add it to the test?
```diff
diff --git a/libpod/container_internal_common.go b/libpod/container_internal_common.go
index 1fdca63cf..72564254a 100644
--- a/libpod/container_internal_common.go
+++ b/libpod/container_internal_common.go
@@ -573,6 +573,10 @@ func (c *Container) generateSpec(ctx context.Context) (s *spec.Spec, cleanupFunc
                g.AddAnnotation("org.systemd.property.TimeoutStopUSec", annotation)
        }
 
+       if c.config.StopSignal != 0 {
+               g.AddAnnotation("org.systemd.property.KillSignal", strconv.FormatUint(uint64(c.config.StopSignal), 10))
+       }
+
        if _, exists := g.Config.Annotations[annotations.ContainerManager]; !exists {
                g.AddAnnotation(annotations.ContainerManager, annotations.ContainerManagerLibpod)
        }
```

Comment (giuseppe) [reply libpod/container_internal_common.go:577]: thanks!  I've amended the change and the test

**Review by Luap99 — APPROVED** [review]
LGTM

**Comment by mheon** [comment]
LGTM

**Comment by giuseppe** [comment]
good to merge?

**Pull request merged**